tasks/upload.py

```diff
@@ -570,65 +550,51 @@ def schedule_task(
         argument_list: list[dict],
         commit_report: CommitReport,
         upload_context: UploadContext,
-        db_session,
-        checkpoints=None,
+        db_session: Session,
+        checkpoints: CheckpointLogger | None,
     ):
         commit_yaml = commit_yaml.to_dict()
 
-        res = None
-        if (
-            commit_report.report_type is None
-            or commit_report.report_type == ReportType.COVERAGE.value
-        ):
-            res = self._schedule_coverage_processing_task(
+        if upload_context.report_type == ReportType.COVERAGE:
+            assert (
+                commit_report.report_type is None
+                or commit_report.report_type == ReportType.COVERAGE.value
+            )
```

Comment:
I’m still a bit confused what the relationship between `upload_context.report_type`, `commit_report.report_type` and `report_code` is?

Reply:
I think `upload_context.report_type` is the type the API tells the worker we should be processing.

`commit_report.report_type` is the report type for that `commit_report` (saved in the DB)

`report_code` is something different... it was intended to be used with local uploads, but I don't think anyone currently uses it. The code is most of the time "default" (I think), but using the CLI you can upload a report and give it a different code. This way you can check coverage without being impacted by other uploads (from CI for example)

Reply:
That is interesting. My confusion primarily comes from here:
- The code was using `upload_context.report_type` for creating the `XYZReportService`
- The `commit_report` is being created using only the `upload_context.report_code`
- The old code was then matching on `commit_report.report_type` for scheduling the tasks, which I changed.

Therefore, I believe that all 3 variables actually mean the same thing. Or phrased the other way around: the code would be broken in surprising ways if they didn’t.